### PR TITLE
fix: node should return duplicated when tx_hash collision

### PR DESCRIPTION
**Pull request by u2**
https://github.com/nervosnetwork/ckb/pull/1718#issuecomment-542601807

### CODE REVIEW

File: tx-pool/src/component/container.rs [hunk 1]
```diff
@@ -202,6 +202,13 @@ impl SortedTxMap {
         self.entries.get(id)
     }
 
+    pub fn get_sorted_key(&self, id: &ProposalShortId) -> Option<AncestorsScoreSortKey> {
```
Comment (jjyr) [inline tx-pool/src/component/container.rs]: Please delete this function.

File: tx-pool/src/process/submit_txs.rs [hunk 2]
```diff
@@ -242,6 +242,11 @@ fn resolve_tx<'a>(
     }
 
     let short_id = tx.proposal_short_id();
+
+    if tx_pool.contains_proposal_id(&short_id) {
```
Comment (jjyr) [inline tx-pool/src/process/submit_txs.rs]: Can you move this condition to `process_txs`.

File: tx-pool/src/component/container.rs (hunk 1 above)
Comment (u2) [reply tx-pool/src/component/container.rs]: done.

File: tx-pool/src/process/submit_txs.rs (hunk 2 above)
Comment (u2) [reply tx-pool/src/process/submit_txs.rs]: I move it in PreResolveTxsProcess

**Comment by jjyr** [comment]
cherry pick to https://github.com/nervosnetwork/ckb/pull/1736

**Pull request closed without merging**